.github/workflows/main.yml

```diff
@@ -0,0 +1,85 @@
+name: Build and test
+
+on:
+  push:
+  pull_request:
+  schedule:
+    # run daily, this refreshes the cache
+    - cron: '13 2 * * *'
+
+jobs:
+  python-test:
+    name: Python tests
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Run python tests
+        run: bash .github/python-nosetests.sh
+
+  ocaml-test:
+    name: Ocaml tests
+    runs-on: ubuntu-20.04
+    env:
+      package: "xapi-xenopsd xapi-xenopsd-simulator xapi-xenopsd-cli" # xapi-xenopsd-xc
+      modules: "xapi-xenopsd,xapi-xenopsd-simulator,xapi-xenopsd-cli" # xapi-xenopsd-xc
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Pull configuration from xs-opam
+        run: |
+          curl --fail --silent https://raw.githubusercontent.com/xapi-project/xs-opam/master/tools/xs-opam-ci.env | cut -f2 -d " " > .env
+
+      - name: Load environment file
+        id: dotenv
+        uses: falti/dotenv-action@v0.2.4
+
+      - name: Retrieve date for cache key
+        id: cache-key
+        run: echo "::set-output name=date::$(/bin/date -u "+%Y%m%d")"
+        shell: bash
+
+      - name: Restore opam cache
+        id: opam-cache
+        uses: actions/cache@v2
+        with:
+          path: "~/.opam"
+          # invalidate cache daily, gets built daily using a scheduled job
+          key: ${{ steps.cache-key.outputs.date }}
+
+      - name: Use ocaml
+        uses: avsm/setup-ocaml@v1
```

Comment:
Should we not use `avsm/setup-ocaml@master` here to make sure we get updates?

Reply:
We moved from using a specific version (e.g. v1.1.0) to v1, which I think ensures we get updates

Reply:
Looks good to me since `avsm/setup-ocam` is on the `v1` track.